### ISSUE TITLE
feature / subprojects

### ISSUE THREAD

**Issue by MikeMitterer** [google/grinder.dart#267]
Maybe it's a bit early for grinder but I think we'll need subprojects in the future. As you can see in my SS all my samples have their own pubspec.yaml - and yes - this is necessary. The more samples you provide the more complicate it becomes to control the samples via parent pubspec... 

![screenshot-1790](https://cloud.githubusercontent.com/assets/116654/8010680/78bc2db2-0bb2-11e5-9b88-e75319adcbca.png)

**Comment by seaneagan** [comment]
I think this works already by adding a dependency on grinder to your sub-projects pubspecs, and giving them each their own `tool/grind.dart` script:

```
cd example/mdl_card
pub global activate den
den install grinder
pub get
pub run grinder:init
# edit `example/mdl_card/tool/grind.dart` as necessary
grind
```

Were you requesting something else?

**Comment by zoechi** [comment]
I do something like
```
projects.forEach((p) {
    Pub.run('grinder',
          arguments: ['check'],
          runOptions: new RunOptions(workingDirectory: p));
});
```

**Comment by zoechi** [comment]
@MikeMitterer I tried it with a directory structure like yours but `pub get`/`pub upgrade` ruinded all symlinks within `examples`. Using another directory (none of the official top-level directory names) fixed it. I wonder if you run into this as well?

**Comment by MikeMitterer** [comment]
@zoechi Yes I had no luck either - I'm using a combination of my own buildSample-dart-script (not public) and a good old bash-script for this. It jumps into each sample folder runs the necessary tasks jumps back and continues with the next sample folder. 

Works nice but I'd like to switch all these tasks to grinder so I made this feature-request...

**Comment by zoechi** [comment]
Grinder works nice the way I showed above, just `pub xxx` messes with subrojects when they are in the `example` folder - just to be clear :)
